File: Java/Unit 01/Lec 04/Q3.java
Single File Programming Question
Problem Statement



Sam, the esteemed captain of his sports team, desires to write a program that displays the name of his team. The program should prompt Sam for the team’s name and subsequently present it in a refined manner.



Can you assist Sam in this task?

Input format :
The input consists of a string, which can include letters, numbers, spaces and special characters.

Output format :
The output prints "Sam's team name is: " followed by the team name.



Refer to the sample output for formatting specifications.

Code constraints :
The given testcases fall under the following constraints:

The length of the team's name should be between 1 and 100 characters.

Sample test cases :
Input 1 :
Tigers
Output 1 :
Sam's team name is: Tigers
Input 2 :
The Winners 2024!
Output 2 :
Sam's team name is: The Winners 2024!


Code



import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String teamName = sc.nextLine();
        System.out.println("Sam's team name is: " + teamName);
    }
}

File: CPP /Unit 05/Lec 31/Q3.cpp
Single File Programming Question
Problem Statement



Nila is organizing a local community event, and she has received a list of names from people who registered online. Since the number of registrants can vary, Nila wants to dynamically allocate memory to store the names rather than using a fixed array size. Once the names are entered, Nila needs to sort them in alphabetical order to better manage the registrations.



Your task is to help Nila by writing a program that first allocates memory dynamically for the names, sorts them alphabetically, and then displays the sorted list of names.

Input format :
The first line contains an integer n, representing the number of names Nila needs to sort.

The next n lines contain one name each (a string), representing the names of the people who registered.

Output format :
The output prints the sorted list of names in alphabetical order, separated by spaces.



Refer to the sample output for the formatting specifications.

Code constraints :
1 ≤ n ≤ 25

1 ≤ length of each student name ≤ 15

Sample test cases :
Input 1 :
5
Sharma
Dev
Ankit
Samir
Robert
Output 1 :
Ankit Dev Robert Samir Sharma

Whitelist
Set 1:
new
delete


Code


#include <iostream>
#include <string>
using namespace std;

int main() {
    int n;
    cin >> n;

    string* names = new string[n];

    for (int i = 0; i < n; i++) {
        cin >> names[i];
    }

    for (int i = 0; i < n - 1; i++) {
        for (int j = i + 1; j < n; j++) {
            if (names[i] > names[j]) {
                string temp = names[i];
                names[i] = names[j];
                names[j] = temp;
            }
        }
    }

    for (int i = 0; i < n; i++) {
        cout << names[i];
        if (i != n - 1) cout << " ";
    }

    delete[] names;
    return 0;
}
